### DISCUSSION TITLE
Accessing ImageMagick._api & its documentation?

### DISCUSSION THREAD

**Discussion by shubhendusaurabh** [dlemstra/magick-wasm#23] [Q&A] [answered]
Hi, 
I want to use other filters like vignette, adaptive scale etc.

Going through source code i can see charcoal filter uses internal api `ImageMagick._api._MagickImage_Charcoal` method.

How can i access the api in JS & its documentation about other available methods ?

Thanks!

**Comment by dlemstra** [comment, marked as answer]
There are probably some tricks to access this but the goal of this library is to hide the internals of that API. Feel free to open an issue to request certain methods to be added or feel free to open up a pull request to implement those methods in this library. But or the API you will only have a typescript definition file and the source code that can be found here: https://github.com/dlemstra/Magick.Native

**Reply by shubhendusaurabh** [reply]
Thanks, i'll look into native source code.

could internal api be exposed till all other methods are pending implementation?
when a stable release is done or goal is reached, it could be removed.

**Reply by dlemstra** [reply]
If you want to use that api you will need to do the same things as adding it to the library. It would help me more if that code was added to the library instead.

**Comment by dlemstra** [comment]
Here is an example for the vignette method that will become available in the next release: https://github.com/dlemstra/magick-wasm/commit/ed3bc9e154db6f1a5faf033f142ff1228fd31719
